custom/icds_reports/static/js/directives/cas-export/cas-export.directive.js

```diff
@@ -129,6 +130,24 @@ function CasExportController($window, $location, locationHierarchy, locationsSer
         return locationsService.getLocations(level, locationsCache, vm.selectedLocations, true);
     };
 
+    vm.userHaveAccessToAllLocations = function (locations) {
+        var haveAccessToAllLocationsForLevel = true;
+        window.angular.forEach(locations, function (location) {
+            if (!location.user_have_access) {
+                haveAccessToAllLocationsForLevel = false;
+            }
+        });
+        return haveAccessToAllLocationsForLevel;
+    };
+
+    vm.userLocationIdIsNull = function () {
+        return ["null", "undefined"].indexOf(vm.userLocationId) !== -1;
```

Comment:
Do you want to check null and undefined? if yes then doing it that way would be better ```vm.userLocationId === null || vm.userLocationId === undefined```

Reply:
following the convention from download directive

Reply:
can you point out the same? because if you are doing the former it might resolve the same as string not as actual null or undefined.

Reply:
https://github.com/challabeehyv/commcare-hq/blob/cr/bf_birth/custom/icds_reports/static/js/directives/download/download.directive.js#L227

They have deliberately put it that way which means the string values are coming as null or undefined. so thought of leaving it like that since both uses same method

Reply:
Sounds weird tagging @czue if he got some insight about this

Reply:
agree with shubham that this is a little odd (also seems slightly different unless its converting `null` and `undefined` to strings silently somewhere), but if this is the convention elsewhere its probably fine

Reply:
I don't have any familiarity with this pattern or why it would be used. Looks like @Rohit25negi is the author of the original code that was used - maybe he has some insight? https://github.com/challabeehyv/commcare-hq/commit/e8e9a69b0a844a60d8d8f8a052467f3115269030#diff-57a6e3501ebaf4aff43ac70619c535a6R207-R209

Reply:
Yeah, I touched them when working on the Tabular report location filters. But only used what was being used. Not super confident about when it was started and where it may be silently converting them to the strings.
I think its the pattern other places we have been following. fine to keep it. Would be great if to also include actual `null` and `undefined` in the list

Reply:
added